### PR TITLE
0.2.1 sovity

### PR DESCRIPTION
**Pull request by ununhexium**

### CODE REVIEW

File: extensions/control-plane/transfer/transfer-data-plane/src/main/java/org/eclipse/edc/connector/transfer/dataplane/flow/ProviderPushTransferDataFlowController.java [hunk 1]
```diff
@@ -64,6 +80,14 @@ private DataFlowRequest createRequest(DataRequest dataRequest, DataAddress sourc
                 .destinationType(dataRequest.getDestinationType())
                 .destinationDataAddress(dataRequest.getDataDestination())
                 .callbackAddress(callbackUrl != null ? callbackUrl.get() : null)
+                .properties(parameterizations)
                 .build();
     }
+
+    private static void extractIfPresent(DataRequest dataRequest, String key, Map<String, String> parameterizations) {
```
Comment (efiege) [inline extensions/control-plane/transfer/transfer-data-plane/src/main/java/org/eclipse/edc/connector/transfer/dataplane/flow/ProviderPushTransferDataFlowController.java:87]: Maybe return DataRequest

File: extensions/control-plane/transfer/transfer-data-plane/src/main/java/org/eclipse/edc/connector/transfer/dataplane/flow/ProviderPushTransferDataFlowController.java [hunk 2]
```diff
@@ -64,6 +80,14 @@ private DataFlowRequest createRequest(DataRequest dataRequest, DataAddress sourc
                 .destinationType(dataRequest.getDestinationType())
                 .destinationDataAddress(dataRequest.getDataDestination())
                 .callbackAddress(callbackUrl != null ? callbackUrl.get() : null)
+                .properties(parameterizations)
```
Comment (efiege) [inline extensions/control-plane/transfer/transfer-data-plane/src/main/java/org/eclipse/edc/connector/transfer/dataplane/flow/ProviderPushTransferDataFlowController.java:83]: Maybe have a look at if privateProperties can be used. We for sure have to migrate this in future core-edc versions

File: extensions/control-plane/transfer/transfer-data-plane/src/test/java/org/eclipse/edc/connector/transfer/dataplane/flow/ProviderPushTransferDataFlowControllerTest.java [hunk 3]
```diff
@@ -115,6 +120,40 @@ void verifyTransferSuccessWithAdditionalProperties() {
...
+    /**
+     * Sovity workaround for provider push asset parameterization that got removed after EDC MS8 when changing the protocol to the IDS one.
+     */
+    @Test
+    void canUseWorkaroundToPassProviderProxyParameters() {
+        var source = testDataAddress();
+        var destination = DataAddress.Builder.newInstance()
+                .type("test-type")
+                .properties(Map.of(
+                        "https://sovity.de/workaround/proxy/param/" + METHOD, "METHOD",
```
Comment (efiege) [inline extensions/control-plane/transfer/transfer-data-plane/src/test/java/org/eclipse/edc/connector/transfer/dataplane/flow/ProviderPushTransferDataFlowControllerTest.java:132]: Reuse the constant

File: extensions/control-plane/transfer/transfer-data-plane/src/main/java/org/eclipse/edc/connector/transfer/dataplane/flow/ProviderPushTransferDataFlowController.java (hunk 2 above)
Comment (ununhexium) [reply extensions/control-plane/transfer/transfer-data-plane/src/main/java/org/eclipse/edc/connector/transfer/dataplane/flow/ProviderPushTransferDataFlowController.java:83]: Using `properties` allows me to keep the rest of the codebase the same as this is the field accessed in `BaseSourceHttpParamsDecorator` to do the assets param ; thus reducing the code changes and easing the migration to subsequent versions.

File: extensions/control-plane/transfer/transfer-data-plane/src/test/java/org/eclipse/edc/connector/transfer/dataplane/flow/ProviderPushTransferDataFlowControllerTest.java (hunk 3 above)
Comment (ununhexium) [reply extensions/control-plane/transfer/transfer-data-plane/src/test/java/org/eclipse/edc/connector/transfer/dataplane/flow/ProviderPushTransferDataFlowControllerTest.java]: yes

https://github.com/sovity/core-edc/blob/sovity/0.2.1.x/extensions/control-plane/transfer/transfer-data-plane/src/test/java/org/eclipse/edc/connector/transfer/dataplane/flow/ProviderPushTransferDataFlowControllerTest.java#L129

**Pull request closed without merging**